modules/src/core/ics02_client/context.rs

```diff
@@ -78,6 +80,84 @@ pub trait ClientReader {
     fn client_counter(&self) -> Result<u64, Error>;
 }
 
+/// Defines a subset of the `ClientReader`'s methods that a light-client implementation can access.
+/// A blanket implementation of this trait is provided for all types that implement `ClientReader`.
+///
+/// Note: This trait is not a supertrait of `ClientReader` because it uses trait objects and cannot
+/// depend on `AnyClientState` and `AnyConsensusState` due to a circular dependency problem.
+pub trait LightClientReader {
+    fn consensus_state(
+        &self,
+        client_id: &ClientId,
+        height: Height,
+    ) -> Result<Box<dyn ConsensusState>, Error>;
+
+    /// Similar to `consensus_state`, attempt to retrieve the consensus state,
+    /// but return `None` if no state exists at the given height.
+    fn maybe_consensus_state(
```

Comment:
This is only used with the tendermint light client, and the logic could easily be moved there. I would vote for keeping the traits as simple as possible and removing this method.

Reply:
Good point! Done https://github.com/informalsystems/ibc-rs/pull/2332/commits/65e4ed41cbb71c6261a5d11a1971625cb2e3059a